Index: gnome-screenshot-3.8.3/src/screenshot-application.c
===================================================================
--- gnome-screenshot-3.8.3.orig/src/screenshot-application.c	2013-12-16 10:54:10.152384553 +1300
+++ gnome-screenshot-3.8.3/src/screenshot-application.c	2013-12-16 10:54:10.148384553 +1300
@@ -139,7 +139,8 @@
           !self->priv->should_overwrite)
         {
           gchar *folder = screenshot_dialog_get_folder (dialog);
-          gchar *folder_name = g_path_get_basename (folder);
+          gchar *folder_uri = g_path_get_basename (folder);
+          gchar *folder_name = g_uri_unescape_string (folder_uri, NULL);
           gchar *file_name = screenshot_dialog_get_filename (dialog);
           gchar *detail = g_strdup_printf (_("A file named \"%s\" already exists in \"%s\""),
                                            file_name, folder_name);
@@ -153,6 +154,7 @@
 
           g_free (folder);
           g_free (folder_name);
+          g_free (folder_uri);
           g_free (file_name);
           g_free (detail);
 
